src/CarbonAware.LocationSources/src/LocationSource.cs

```diff
@@ -0,0 +1,137 @@
+using CarbonAware.Model;
+using CarbonAware.Interfaces;
+using CarbonAware.Exceptions;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using Microsoft.Extensions.Options;
+using CarbonAware.LocationSources.Configuration;
+
+namespace CarbonAware.LocationSources;
+
+/// <summary>
+/// Represents a location source.
+/// </summary>
+public class LocationSource : ILocationSource
+{
+    private readonly ILogger<LocationSource> _logger;
+
+    private IDictionary<string, NamedGeoposition>? _namedGeopositions;
+
+    private static readonly JsonSerializerOptions options = new JsonSerializerOptions(JsonSerializerDefaults.Web);
+
+    private IOptionsMonitor<LocationDataSourcesConfiguration> _configurationMonitor { get; }
+
+    private LocationDataSourcesConfiguration _configuration => _configurationMonitor.CurrentValue;
+
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="LocationSource"/> class.
+    /// </summary>
+    /// <param name="logger">The logger for the LocationSource</param>
+    public LocationSource(ILogger<LocationSource> logger, IOptionsMonitor<LocationDataSourcesConfiguration> monitor)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _configurationMonitor = monitor ?? throw new ArgumentNullException(nameof(monitor));
+    }
+
+    public async Task<Location> ToGeopositionLocationAsync(Location location)
+    {
+        switch (location.LocationType)
```

Comment:
Do we need `LocationType` anymore?  I think we can simplify and remove a lot of code if `Location` is just a `Name`, `Latitude`, and `Longitude`.

Reply:
Sure. I'll do more cleanup 👍